agent/agents/postgres/pgstatstatements/pgstatstatements_test.go

```diff
@@ -440,3 +440,127 @@ func TestPGStatStatementsQAN(t *testing.T) {
 		assert.LessOrEqual(t, actual.Postgresql.MSharedBlkReadTimeSum, actual.Common.MQueryTimeSum)
 	})
 }
+
+func TestPGStatStatementsQPS(t *testing.T) {
+	sqlDB := tests.OpenTestPostgreSQL(t)
+	defer sqlDB.Close() //nolint:errcheck
+	db := reform.NewDB(sqlDB, postgresql.Dialect, nil)
+
+	_, err := db.Exec("CREATE EXTENSION IF NOT EXISTS pg_stat_statements SCHEMA public")
+	require.NoError(t, err)
+
+	defer func() {
+		_, err := db.Exec("DROP EXTENSION pg_stat_statements")
+		assert.NoError(t, err)
+	}()
+
+	// filterInsertQueries retrieves only buckets for insert queries used by test.
+	filterInsertQueries := func(mb []*agentv1.MetricsBucket) []*agentv1.MetricsBucket {
+		res := make([]*agentv1.MetricsBucket, 0, len(mb))
+		for _, b := range mb {
+			switch {
+			case strings.Contains(b.Common.Fingerprint, "insert /* controller='test' */"):
+				res = append(res, b)
+			default:
+				continue
+			}
+		}
+		return res
+	}
+
+	t.Run("check query count for low cache size", func(t *testing.T) {
+		p := setup(t, db, minPgSSCacheSize)
+		runTimes := 5000
+
+		defer func() {
+			for i := 0; i < runTimes; i++ {
+				_, err = db.Exec(fmt.Sprintf("drop table if exists t%d", i))
+				require.NoError(t, err)
+			}
+		}()
+		for i := 0; i < runTimes; i++ {
+			_, err = db.Exec(fmt.Sprintf("create /* controller='test' */ table t%d (id int);", i))
+			require.NoError(t, err)
+			_, err = db.Exec(fmt.Sprintf("insert /* controller='test' */ into t%d values(1);", i))
+			require.NoError(t, err)
+		}
+
+		buckets, err := p.getNewBuckets(context.Background(), time.Date(2019, 4, 1, 10, 59, 0, 0, time.UTC), 60)
+		require.NoError(t, err)
+		insertBuckets := filterInsertQueries(buckets)
+		mismatchedCount := 0
+		for _, b := range insertBuckets {
+			assert.Equal(t, float32(1), b.Common.NumQueries)
+			if b.Common.NumQueries != 1 {
+				mismatchedCount++
+			}
+		}
+		assert.Zero(t, mismatchedCount)
+
+		// re-run insert queries and check that there are now mismatches since the cache can't hold all the previous bucket.
+		for i := 0; i < runTimes; i++ {
+			_, err = db.Exec(fmt.Sprintf("insert /* controller='test' */ into t%d values(1);", i))
+			require.NoError(t, err)
+		}
+		buckets, err = p.getNewBuckets(context.Background(), time.Date(2019, 4, 1, 10, 59, 0, 0, time.UTC), 60)
+		require.NoError(t, err)
+		insertBuckets = filterInsertQueries(buckets)
+		mismatchedCount = 0
+		for _, b := range insertBuckets {
+			if b.Common.NumQueries != 1 {
+				mismatchedCount++
+			}
+		}
+		assert.Greater(t, mismatchedCount, 0) //nolint:testifylint // this makes what we're testing more explicit
```

Comment:
we shouldn't expect mismatch count > 0

Reply:
The second test checks that the mismatch count is always 0, but for this case (where the cache size is small), it is possible to get mismatches...

Reply:
let's remove this test case then, it makes no sense

Reply:
yeah dropped now - it was more of a confirmation test for when the bug is present.